azurerm/resource_arm_private_dns_cname_record_test.go

```diff
@@ -0,0 +1,348 @@
+		ctx := testAccProvider.Meta().(*ArmClient).StopContext
+		resp, err := conn.Get(ctx, resourceGroup, zoneName, privatedns.CNAME, aName)
+		if err != nil {
+			return fmt.Errorf("Bad: Get CNAME RecordSet: %+v", err)
+		}
+
+		if resp.StatusCode == http.StatusNotFound {
+			return fmt.Errorf("Bad: Private DNS CNAME record %s (resource group: %s) does not exist", aName, resourceGroup)
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureRMPrivateDnsCNameRecordDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*ArmClient).privateDns.RecordSetsClient
+	ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_private_dns_cname_record" {
+			continue
+		}
+
+		aName := rs.Primary.Attributes["name"]
+		zoneName := rs.Primary.Attributes["zone_name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+
+		resp, err := conn.Get(ctx, resourceGroup, zoneName, privatedns.A, aName)
+
+		if err != nil {
+			if resp.StatusCode == http.StatusNotFound {
+				return nil
+			}
+
+			return err
+		}
+
+		return fmt.Errorf("Private DNS CNAME record still exists:\n%#v", resp.RecordSetProperties)
+	}
+
+	return nil
+}
+
+func testAccAzureRMPrivateDnsCNameRecord_basic(rInt int, location string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_private_dns_zone" "test" {
+  name                = "acctestzone%d.com"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+}
+
+resource "azurerm_private_dns_cname_record" "test" {
+  name                = "mycnamerecord%d"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  zone_name           = "${azurerm_private_dns_zone.test.name}"
+  ttl                 = 300
+  record              = "contoso.com"
+}
+`, rInt, location, rInt, rInt)
+}
+
+func testAccAzureRMPrivateDnsCNameRecord_requiresImport(rInt int, location string) string {
+	template := testAccAzureRMPrivateDnsCNameRecord_basic(rInt, location)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_private_dns_cname_record" "import" {
+  name                = "${azurerm_private_dns_cname_record.test.name}"
+  resource_group_name = "${azurerm_private_dns_cname_record.test.resource_group_name}"
+  zone_name           = "${azurerm_private_dns_cname_record.test.zone_name}"
+  ttl                 = 300
+  record              = "contoso.com"
+}
+`, template)
+}
+
+func testAccAzureRMPrivateDnsCNameRecord_subdomain(rInt int, location string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_private_dns_zone" "test" {
+  name                = "acctestzone%d.com"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+}
+
+resource "azurerm_private_dns_cname_record" "test" {
+  name                = "mycnamerecord%d"
```

Comment:
we have a cleanup tool that works based off the prefix `acctest` - as such could we make this:

```suggestion
  name                = "acctestcname%d"
```